src/deluge/model/mod_controllable/mod_controllable_audio.cpp

```diff
@@ -1503,33 +1512,68 @@ int32_t ModControllableAudio::readTagFromFile(char const* tagName, ParamManagerF
 		storageManager.exitTag("delay");
 	}
 
-	else if (!strcmp(tagName, "compressor")) { // Remember, Song doesn't use this
+	else if (!strcmp(tagName, "AudioCompressor")) {
+		while (*(tagName = storageManager.readNextTagOrAttributeName())) {
+			if (!strcmp(tagName, "attack")) {
+				q31_t masterCompressorAttack = storageManager.readTagOrAttributeValueInt();
+				compressor.setAttack(masterCompressorAttack);
+				storageManager.exitTag("attack");
+			}
+			else if (!strcmp(tagName, "release")) {
+				q31_t masterCompressorRelease = storageManager.readTagOrAttributeValueInt();
+				compressor.setRelease(masterCompressorRelease);
+				storageManager.exitTag("release");
+			}
+			else if (!strcmp(tagName, "thresh")) {
+				q31_t masterCompressorThresh = storageManager.readTagOrAttributeValueInt();
+				compressor.setThreshold(masterCompressorThresh);
+				storageManager.exitTag("thresh");
+			}
+			else if (!strcmp(tagName, "ratio")) {
+				q31_t masterCompressorRatio = storageManager.readTagOrAttributeValueInt();
+				compressor.setRatio(masterCompressorRatio);
+				storageManager.exitTag("ratio");
+			}
+			else if (!strcmp(tagName, "compHPF")) {
+				q31_t masterCompressorSidechain = storageManager.readTagOrAttributeValueInt();
+				compressor.setSidechain(masterCompressorSidechain);
+				storageManager.exitTag("compHPF");
+			}
+			else {
+				storageManager.exitTag(tagName);
+			}
+		}
+		storageManager.exitTag("AudioCompressor");
+	}
+	// this is actually the sidechain but pre 1.1 songs save it as compressor
+	else if (!strcmp(tagName, "compressor") || !strcmp(tagName, "Sidechain")) { // Remember, Song doesn't use this
```

Comment:
Is the  `Remember, Song doesn't use this` comment leftover from when the code was originally copied from somewhere else? It seems like song does in fact use this.

Reply:
It's an old comment but I'm still pretty sure it doesn't - it reads the tags because it inherits from mod controllable but never actually calls the sidechain comp

Reply:
It does use the `compressor` parameters though right?

Reply:
Yeah this is why I renamed the parameter - song uses the real compressor, but not the sidechain envelope (formerly named compressor in the code)